test/test_grids.jl

```diff
@@ -539,21 +544,21 @@ function test_basic_lat_lon_general_grid(FT)
 
     @test typeof(grid_reg.Δzᵃᵃᶜ) == typeof(grid_reg.Δzᵃᵃᶠ) == FT
 
-    @test xspacings(grid_reg, Center(), Center(), with_halos=true) == grid_reg.Δxᶜᶜᵃ
-    @test xspacings(grid_reg, Center(), Face(),   with_halos=true) == grid_reg.Δxᶜᶠᵃ
-    @test xspacings(grid_reg, Face(),   Center(), with_halos=true) == grid_reg.Δxᶠᶜᵃ
-    @test xspacings(grid_reg, Face(),   Face(),   with_halos=true) == grid_reg.Δxᶠᶠᵃ
-    @test yspacings(grid_reg, Center(), Face(),   with_halos=true) == grid_reg.Δyᶜᶠᵃ
-    @test yspacings(grid_reg, Face(),   Center(), with_halos=true) == grid_reg.Δyᶠᶜᵃ
-    @test zspacings(grid_reg, Center(), with_halos=true) == grid_reg.Δzᵃᵃᶜ
-    @test zspacings(grid_reg, Face(),   with_halos=true) == grid_reg.Δzᵃᵃᶠ
-
-    @test xspacings(grid_reg, Center(), Center(), Center()) == xspacings(grid_reg, Center(), Center())
-    @test xspacings(grid_reg, Face(),   Face(),   Center()) == xspacings(grid_reg, Face(),   Face())
-    @test yspacings(grid_reg, Center(), Face(),   Center()) == yspacings(grid_reg, Center(), Face())
-    @test yspacings(grid_reg, Face(),   Center(), Center()) == yspacings(grid_reg, Face(),   Center())
-    @test zspacings(grid_reg, Face(),   Face(),   Center()) == zspacings(grid_reg, Center())
-    @test zspacings(grid_reg, Face(),   Center(), Face()  ) == zspacings(grid_reg, Face())
+    @test all(xspacings_field(grid_reg, Center(), Center()) .== reshape(grid_reg.Δxᶜᶜᵃ[1:Nφ],   1, Nφ,   1))
+    @test all(xspacings_field(grid_reg, Center(), Face()  ) .== reshape(grid_reg.Δxᶜᶠᵃ[1:Nφ+1], 1, Nφ+1, 1))
+    @test all(xspacings_field(grid_reg, Face(),   Center()) .== reshape(grid_reg.Δxᶠᶜᵃ[1:Nφ],   1, Nφ,   1))
+    @test all(xspacings_field(grid_reg, Face(),   Face())   .== reshape(grid_reg.Δxᶠᶠᵃ[1:Nφ+1], 1, Nφ+1, 1))
+    @test all(yspacings_field(grid_reg, Center(), Face()) .== grid_reg.Δyᶜᶠᵃ)
+    @test all(yspacings_field(grid_reg, Face(),   Center()) .== grid_reg.Δyᶠᶜᵃ)
+    @test all(zspacings_field(grid_reg, Center()) .== grid_reg.Δzᵃᵃᶜ)
+    @test all(zspacings_field(grid_reg, Face()) .== grid_reg.Δzᵃᵃᶠ)
```

Comment:
is `_field` just needed for testing? It might be better to implement `.==` and `==` for operations...

Reply:
As a very quick start I would suggest something like

```julia
function Base.==(op1::AbstractOperation, op2::AbstractOperation)
    f1 = compute!(Field(op1))
    f2 = compute!(Field(op2))
    return f1 == f2
end
```

plus various permutations for comparing operations to fields, etc.

Reply:
Yeah it was just for testing but overloading `Base.==` makes sense here. I'll give it a try.

Reply:
you could put a `TODO:` comment to avoid the memory allocation and use a kernel instead too

Reply:
Apparently `.==` already works for `KernelFunctionOperator`s so I didn't have to add any new code! The magic of Julia :mage: